techniques/fileDistribution/fileTemplate/1.0/fileTemplate.st

```diff
@@ -180,13 +194,13 @@ pass3::
       #Case with execution
       
       "any" usebundle => rudder_common_reports_generic("fileTemplate", "command_execution_${posthook_canon[${index}]}", "${trackingkey[${index}]}", "Posthook", "${dst[${index}]}", "The command ${posthook[${index}]} from postHook execution "),
-      ifvarclass => "posthook_launch_${index}";
+      ifvarclass => "command_execution_${posthook_canon[${index}]}_reached";
 
 
       #Case without execution
 
       "report_${index}" usebundle  => rudder_common_report("fileTemplate", "result_na", "${trackingkey[${index}]}", "Posthook", "${dst[${index}]}", "No post-modification needed to run"),
-      ifvarclass => "!file_modified_${index}.posthook_specified_${index}";
+      ifvarclass => "!file_modified_${index}.posthook_specified_${index}.!command_execution_${posthook_canon[${index}]}_reached";
```

Comment:
A class can only be defined or not defined.
A class can be not defined, be be defined on next pass.
That's why using negation in class expressions is touchy.

But i see that this is enclosed in pass3, so there is no problem.